frontend/libs/studio-components/src/components/StudioTextField/StudioTextfieldSchema/StudioTextfieldSchema.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useMemo, useState } from 'react';
+import {
+  StudioToggableTextfield,
+  type StudioToggableTextfieldProps,
+} from '../StudioToggableTextfield';
+import type { JsonSchema } from '../../../../../../packages/shared/src/types/JsonSchema';
+import { useTranslation } from 'react-i18next';
+import { StudioJSONValidatorUtils } from '../StudioJSONValidatorUtils';
+
+export type StudioTextfieldSchemaProps = {
+  schema: JsonSchema;
+  propertyPath: string;
+  jsonValidator: any;
+} & StudioToggableTextfieldProps;
+
+export const StudioTextfieldSchema = ({
+  jsonValidator,
+  schema,
+  inputProps,
+  propertyPath,
+  ...rest
+}: StudioTextfieldSchemaProps) => {
+  const [errorMessage, setErrorMessage] = useState<string | null>();
+  const studioJSONValidator = new StudioJSONValidatorUtils(jsonValidator);
+  const { t } = useTranslation();
+
+  const propertyId = useMemo(
+    () => (schema && propertyPath ? `${schema.$id}#/${propertyPath}` : null),
+    [schema, propertyPath],
+  );
+
+  const validateAgainstSchema = (event: React.ChangeEvent<HTMLInputElement>): string | null => {
+    const newValue = event.target.value;
+    if (studioJSONValidator.isPropertyRequired(schema, propertyPath) && newValue?.length === 0) {
+      return t('validation_errors.required');
```

Comment:
We've discussed within the team previously about removing direct dependencies on texts inside studio/components, opting instead to pass texts as arguments. What do you think about returning error codes here instead, similar to how the old FormField used to do? See the next comment for more details. 😊